### PR TITLE
add launch context functionality

### PR DESCRIPTION
**Pull request by maimoonak**
Fixes #[issue number]

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [x] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [x] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [x] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/test/java/org/smartregister/fhircore/engine/FhirExtractionTest.kt
Comment (pld) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/FhirExtractionTest.kt:1]: 👍

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt [hunk 1]
```diff
@@ -213,6 +214,21 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
         if (!questionnaireConfig.type.isDefault()) {
           setQuestionnaireResponse(questionnaireResponse.encodeResourceToString())
         }
+        questionnaireConfig.resourceIdentifier?.takeIf { it.isNotBlank() }?.let {
+          val resourceId = IdType(it)
+          val resourceType =
+            resourceId.resourceType?.let { ResourceType.fromCode(it) }
+              ?: questionnaireConfig.resourceType ?: ResourceType.Patient
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt:221]: k so this will default to assuming it's a patient, that seems OK, any implications of this we should be concerned about / keep an eye on for the future?

**Review by pld — APPROVED** [review]
